### PR TITLE
Unity benchmarking Service Charter

### PR DESCRIPTION
**Pull request by TimothyTickle**
~**December 8**: Last call for community reviews~
**January 9**: Last call for oversight reviews

### CODE REVIEW

File: charters/UNITY/charter.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+# Unity Benchmarking Service
+
+## Description
+
+The Unity benchmarking service automates the process of running and benchmarking pipelines.
+
+## Objectives
+
+The Unity benchmarking service aims to automate benchmarking for HCA pipelines, enforce standards for consistent reporting for historic comparison and enable the public to participate in suggesting improvements to pipelines by empowering them to implement improvements and test results using the same infrastructure as the HCA data processing pipelines team. The Unity benchmarking service will be used by those making pipelines for the HCA data processing service and will be public, so anyone can benchmark their implemented ideas against HCA pipelines.
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: What is meant by _the public_?

File: charters/UNITY/charter.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+* Benchmarking for HCA pipelines will be automated and included in Unity
+* Benchmarking data will be available through Unity but not controllable by the users
+* Benchmarking data can change without notice but will be consistent in format and schema with previous data or notice will be given to users
+* Leaderboards showing how pipelines differ will be created as different pipelines for the same assay are collected
+* Documentation on how to use Unity and the assumptions pipelines must meet to be ran using Unity
+* Documentation on benchmarking process and outputs
+
+### Scientific "guardrails"
+
+* Benchmarking pipelines are reviewed by the HCA project and treated as data processing pipelines (engineering updates do not require HCA oversight; scientific updates do).
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: What is the _HCA project_?

File: charters/UNITY/charter.md (hunk 1 above)
Comment (brianraymor) [inline charters/UNITY/charter.md]: What are _HCA pipelines_ vs _DCP pipelines_?

File: charters/UNITY/charter.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
+# Unity Benchmarking Service
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: Is _Unity_ a codename or an aspiration? Could the directory be named _UnityBenchmarkingService_ instead of _UNITY_?

File: charters/UNITY/charter.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+## Description
+
+The Unity benchmarking service automates the process of running and benchmarking pipelines.
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: What is a _pipeline_?

File: charters/UNITY/charter.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+## Objectives
+
+The Unity benchmarking service aims to automate benchmarking for HCA pipelines, enforce standards for consistent reporting for historic comparison and enable the public to participate in suggesting improvements to pipelines by empowering them to implement improvements and test results using the same infrastructure as the HCA data processing pipelines team. The Unity benchmarking service will be used by those making pipelines for the HCA data processing service and will be public, so anyone can benchmark their implemented ideas against HCA pipelines.
+
+## In-scope
+
+* Benchmarking for HCA pipelines will be automated and included in Unity
+* Benchmarking data will be available through Unity but not controllable by the users
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: What is meant by _not controllable by the users_ - does this mean that data is reported to the users or that the users can view the data?

File: charters/UNITY/charter.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+* Benchmarking data can change without notice but will be consistent in format and schema with previous data or notice will be given to users
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: Users will be notified if the format or schema for benchmarking data changes? This is the first reference to a _schema_.

File: charters/UNITY/charter.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+* Leaderboards showing how pipelines differ will be created as different pipelines for the same assay are collected
+* Documentation on how to use Unity and the assumptions pipelines must meet to be ran using Unity
+* Documentation on benchmarking process and outputs
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: Is this where the schemas are defined?

File: charters/UNITY/charter.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+## Objectives
+
+The Unity benchmarking service aims to automate benchmarking for HCA pipelines, enforce standards for consistent reporting for historic comparison and enable the public to participate in suggesting improvements to pipelines by empowering them to implement improvements and test results using the same infrastructure as the HCA data processing pipelines team. The Unity benchmarking service will be used by those making pipelines for the HCA data processing service and will be public, so anyone can benchmark their implemented ideas against HCA pipelines.
+
+## In-scope
+
+* Benchmarking for HCA pipelines will be automated and included in Unity
+* Benchmarking data will be available through Unity but not controllable by the users
+* Benchmarking data can change without notice but will be consistent in format and schema with previous data or notice will be given to users
+* Leaderboards showing how pipelines differ will be created as different pipelines for the same assay are collected
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: _Differences between pipelines for the same assay will be presented in leaderboards._ ?

File: charters/UNITY/charter.md [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+* Documentation on how to use Unity and the assumptions pipelines must meet to be ran using Unity
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: The _assumptions pipelines must meet to be ran_ was hard for me to parse. 

 I especially don't understand the use of _assumptions_ in this context. Are there not requirements? 

Perhaps - _Documentation for the requirements of pipelines to be benchmarked by the Unity Benchmarking Service._

File: charters/UNITY/charter.md [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+### Scientific "guardrails"
+
+* Benchmarking pipelines are reviewed by the HCA project and treated as data processing pipelines (engineering updates do not require HCA oversight; scientific updates do).
+* Metrics and how metrics are used in the leaderboards are reviewed by the HCA project.
+
+## Out-of-scope
+
+* Educational training or user assistance on how to create pipelines
+* Recommendations on how to improve pipelines based on benchmarking
+* Benchmarking community pipelines are not required to be included in the Unity benchmarking service (but are allowed)
```
Comment (brianraymor) [inline charters/UNITY/charter.md]: _Benchmarking ... **is** not required ... but **is** allowed ..._
Can this be clarified? This team will not benchmark community pipelines, but the DCP community may?

File: charters/UNITY/charter.md (hunk 1 above)
Comment (ambrosejcarr) [reply charters/UNITY/charter.md]: What is meant by _implement ... test results_?

File: charters/UNITY/charter.md [hunk 11]
```diff
@@ -0,0 +1,56 @@
+# Unity Benchmarking Service
+
+## Description
+
+The Unity benchmarking service automates the process of running and benchmarking pipelines.
+
+## Objectives
+
+The Unity benchmarking service aims to automate benchmarking for HCA pipelines, enforce standards for consistent reporting for historic comparison and enable the public to participate in suggesting improvements to pipelines by empowering them to implement improvements and test results using the same infrastructure as the HCA data processing pipelines team. The Unity benchmarking service will be used by those making pipelines for the HCA data processing service and will be public, so anyone can benchmark their implemented ideas against HCA pipelines.
```
Comment (ambrosejcarr) [inline charters/UNITY/charter.md]: > so anyone can benchmark their implemented ideas against HCA pipelines

Is this exclusively scoped to assays with existing pipelines, or does the benchmarking service also cover benchmarking pipelines for new assays? 

Can users test components of pipelines, or must they construct full end-to-end pipelines to use this service?

File: charters/UNITY/charter.md (hunk 5 above)
Comment (ambrosejcarr) [reply charters/UNITY/charter.md]: More basic: What's _data_ here?

File: charters/UNITY/charter.md [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+
+## In-scope
+
+* Benchmarking for HCA pipelines will be automated and included in Unity
+* Benchmarking data will be available through Unity but not controllable by the users
+* Benchmarking data can change without notice but will be consistent in format and schema with previous data or notice will be given to users
+* Leaderboards showing how pipelines differ will be created as different pipelines for the same assay are collected
```
Comment (ambrosejcarr) [inline charters/UNITY/charter.md]: Leaderboards is an interesting concept and a big increase in scope, because it implies that we know, for each assay, what it means to assess data quality. My impression is that we often don't have clarity to make those calls concretely and mathematically. How will Unity approach this? 

An alternative, which would be to simply report descriptive statistics about the outputs, could be a less onerous task.

File: charters/UNITY/charter.md [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+* Documentation on how to use Unity and the assumptions pipelines must meet to be ran using Unity
+* Documentation on benchmarking process and outputs
+
+### Scientific "guardrails"
+
+* Benchmarking pipelines are reviewed by the HCA project and treated as data processing pipelines (engineering updates do not require HCA oversight; scientific updates do).
```
Comment (ambrosejcarr) [inline charters/UNITY/charter.md]: I think this statement is overly cautious to the point of being a bit confusing. 
- Benchmarking pipelines, as I understand them, are either (1) glue to test or summarize the outputs of our gold standard or community pipelines, or (2) the gold standard pipelines themselves. 

In either case, there is no new scientific product here outside what is covered in the next line. I would remove this line, and might add a definition for what a benchmarking pipeline is.

File: charters/UNITY/charter.md [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+* Metrics and how metrics are used in the leaderboards are reviewed by the HCA project.
```
Comment (ambrosejcarr) [inline charters/UNITY/charter.md]: If these are being used to _rank_ pipelines, I think these need AWG approval.

File: charters/UNITY/charter.md [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+## Out-of-scope
+
+* Educational training or user assistance on how to create pipelines
+* Recommendations on how to improve pipelines based on benchmarking
+* Benchmarking community pipelines are not required to be included in the Unity benchmarking service (but are allowed)
+
+## Milestones and Deliverables
+
+__2018:__ Infrastructure operational and secure, HCA data processing pipelines team actively using system with a pipeline.   
+__2019:__ System updated to include all pipelines operational as HCA pipelines.
```
Comment (ambrosejcarr) [inline charters/UNITY/charter.md]: Clarify how "Community PIpelines" are interpreted. Are those "HCA pipelines" and therefore a part of this milestone?

File: charters/UNITY/charter.md (hunk 3 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: UNITY is the name of the service, the aspiration, and a part of the international space station (where we get many of our names, eg Skylab). For clarity, the directory can be called UnityBenchMarkingService.

File: charters/UNITY/charter.md (hunk 4 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: changed to "data processing pipelines" which matches the language of the other charters.

File: charters/UNITY/charter.md (hunk 1 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: Updated language, I hope it is clearer.

File: charters/UNITY/charter.md (hunk 2 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: I change to reviewed by HCA scientific leadership. aka AWG.

File: charters/UNITY/charter.md (hunk 1 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: I changed this to benchmarked pipelines in the DCP as opposed to HCA pipelines. Not sure if we will require community pipelines to run in UNITY (it would be nice) but official HCA pipelines should be in the system and be able to be benchmarked by others.

File: charters/UNITY/charter.md (hunk 5 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: These are the data the pipeline runs on during benchmarking. They will not be changeable except by admins to the system to make sure analysis runs are comparable.

File: charters/UNITY/charter.md (hunk 6 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: The UNITY benchmarking service would enforce file formats and file schemas used in the DCP (for instance outputting sequence data in bams and measurements in zarr). Otherwise, the pipelines will not work with the DCP.

File: charters/UNITY/charter.md (hunk 7 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: The schemas are not defined by UNITY, they are enforced by Unity. They are made by those making the data processing pipelines; or in the case of a new community pipeline possibly by the community (if no standard exists in the system for the data they are generating).

File: charters/UNITY/charter.md (hunk 8 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: Updated to "Leaderboards showing how pipelines that process the same assay differ in performance will be created as different pipelines for the same assay are collected"

File: charters/UNITY/charter.md (hunk 9 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: love it, adding directly.

File: charters/UNITY/charter.md (hunk 10 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: We are not obligated to set up benchmarking for every assay everyone wants to submit to benchmark against. It would be too much work. If we can leverage what we have for standard pipelines then community pipelines can just benchmark using what is in the system. But for a new assay, if the pipeline is in our roadmap then we are more likely to set up the assay in the system for community pipelines. (We have to find reference data to run, potentially define outputs, potentially initiate benchmarking pipelines. Depending on how proactive the community members are, this may be a significant task). I will try to clarify.

File: charters/UNITY/charter.md (hunk 11 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: In the first versions of UNITY, we will focus on assays that are processed in the HCA DCP and potentially those in our roadmap. As we learn more about the product we may lessen those constraints.

One must provide an end-to-end pipeline but could copy one of ours and update it with just the changes of interest.

File: charters/UNITY/charter.md (hunk 12 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: If we can not say what pipelines are better than others then fundamentally we can not improve our own pipelines. These are simple leaderboards but would leverage the same metrics that we use to answer if we improved our pipelines.

File: charters/UNITY/charter.md (hunk 13 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: It is definitely within the purview of the AWG to be aware of changes to benchmarking pipelines. It is our shared contract of how we interpret pipeline performance and is more than benchmarking cost and speed but includes judgment on scientific insight (how to use metrics and what metrics to use). Benchmarking pipelines in UNITY will guide the growth and development of the pipelines and affect their science.

Comment (TimothyTickle) [reply charters/UNITY/charter.md]: I will try to clarify.

File: charters/UNITY/charter.md (hunk 14 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: That is what we mean, will update for clarity.

File: charters/UNITY/charter.md (hunk 15 above)
Comment (TimothyTickle) [reply charters/UNITY/charter.md]: This does not include but does not exclude community pipelines. We just need to focus on getting our standardized pipelines in first.

File: charters/UnityBenchmarkingService/charter.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+* Benchmarking data can change without notice but will be consistent in file format and file schema with previous data. If file format or schema do change (due to updates in the DCP or in standardized data processing pipelines) notice will be given to users
+* Leaderboards showing how pipelines that process the same assay differ in performance will be created as different pipelines for the same assay are collected
+* Documentation for the requirements of pipelines to be benchmarked by the Unity Benchmarking Service
+* Documentation on benchmarking process and outputs
+
+### Scientific "guardrails"
+
+* Pipelines benchmarking standardized HCA data processing pipelines are reviewed by HCA scientific leadership (Analysis Working Group) and treated as standardized data processing pipelines. Changes to these specific benchmarking pipelines will directly affect how standardized HCA data processing pipelines are optimized and comment on what are better scientific results produced by the data processing pipeline they benchmark.
+* Metrics and how metrics are used in the leaderboards are reviewed by the HCA scientific leadership (Analysis Working Group).
```
Comment (lauraclarke) [inline charters/UnityBenchmarkingService/charter.md]: Is this a typo? not sure what how metrics means here? This sentence isn't clear

File: charters/UnityBenchmarkingService/charter.md [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+
+## Out-of-scope
+
+* Educational training or user assistance on how to create pipelines
+* Recommendations on how to improve pipelines based on benchmarking
+* Pipelines benchmarking community data processing pipelines are not required to be included in the Unity benchmarking service (but are allowed). Community members may benchmark community pipelines using data and file format standards in the system. If community pipelines process assays not currently processed by standard HCA pipelines or assays not being scoped by the team, setting up the system will require a significant amount of work and will be evaluated given team bandwidth
```
Comment (lauraclarke) [inline charters/UnityBenchmarkingService/charter.md]: This out of scope statement is unclear in phrasing.

I assume what is meant is that the implementation team won't proactively seek out and add arbitrary pipelines to the system but community members can if they want to but that isn't entirely clear from the statement.

**Review by lauraclarke — APPROVED** [review]
I have made some comments which are about phrasing rather than meaning Please take or leave them as you want.

File: charters/UnityBenchmarkingService/charter.md [hunk 18]
```diff
@@ -0,0 +1,59 @@
+# Unity Benchmarking Service
+
+## Description
+
+The Unity benchmarking service automates the process of running and benchmarking data processing pipelines.
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md]: I would always capitalize the name throughout. Unity benchmarking service -> Unity Benchmarking Service. I won't repeat further in the review.

Do you want to reuse any **Definitions** from your other charters?

File: charters/UnityBenchmarkingService/charter.md [hunk 19]
```diff
@@ -0,0 +1,59 @@
...
+
+## Objectives
+
+The Unity benchmarking service aims to automate benchmarking for data processing pipelines, enforce standards for consistent reporting for historic comparison and enable the those external to the DCP implementation team visibility and opportunity to demonstrate improvements to data processing pipelines using the same infrastructure as the HCA data processing pipelines team. The Unity benchmarking service will be used by those making data processing pipelines for the HCA data processing service and will be hosted in a public-facing manner; so anyone can benchmark their data processing pipelines against benchmarked pipelines in the DCP.
+
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md:16]: HCA data processing pipelines team -> HCA DCP Data Processing Pipelines team

Is it _enforce standards_ or _document standards_?

The Unity benchmarking service ~aims to~ automates benchmarking for data processing pipelines, enforces standards for consistent reporting for historic comparison, and ~enable the those external to the DCP implementation team~, creates ~visibility and opportunity~ opportunities for external pipeline developers to demonstrate improvements to data processing pipelines using ~the same infrastructure as the HCA data processing pipelines team.~ a common benchmarking infrastructure. 

The Unity Benchmarking Service ~will be~ is used by ~those making~ developers of data processing pipelines for the HCA Data Processing Pipelines Service and ~will be~ is hosted in a public-facing manner; so anyone can benchmark their data processing pipelines against benchmarked pipelines in the DCP.

File: charters/UnityBenchmarkingService/charter.md [hunk 20]
```diff
@@ -0,0 +1,59 @@
...
+## In-scope
+
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md:18]: It would substitute _is_  or _are_ for every _will be_ for a stronger statement.

File: charters/UnityBenchmarkingService/charter.md [hunk 21]
```diff
@@ -0,0 +1,59 @@
...
+* Benchmarking for standardized HCA data processing pipelines will be automated and included in Unity
+* Data used to run pipelines during benchmarking will be standardized and enforced to promote comparable analysis runs
+* Benchmarking data can change without notice but will be consistent in file format and file schema with previous data. If file format or schema do change (due to updates in the DCP or in standardized data processing pipelines) notice will be given to users
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md]: I don't think that this needs to be called out in scope unless you add is as a (note) to previous bullet - _Data used to run pipelines ..._

File: charters/UnityBenchmarkingService/charter.md [hunk 22]
```diff
@@ -0,0 +1,59 @@
...
+* Leaderboards showing how pipelines that process the same assay differ in performance will be created as different pipelines for the same assay are collected
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md]: Editorial suggestion:

_When multiple pipelines processing the same assay are collected, leaderboards are created to compare performance._

File: charters/UnityBenchmarkingService/charter.md [hunk 23]
```diff
@@ -0,0 +1,59 @@
...
+* Documentation for the requirements of pipelines to be benchmarked by the Unity Benchmarking Service
+* Documentation on benchmarking process and outputs
+
+### Scientific "guardrails"
+
+* Pipelines benchmarking standardized HCA data processing pipelines are reviewed by HCA scientific leadership (Analysis Working Group) and treated as standardized data processing pipelines. Changes to these specific benchmarking pipelines will directly affect how standardized HCA data processing pipelines are optimized and comment on what are better scientific results produced by the data processing pipeline they benchmark.
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md]: What does _treated as standardized data processing pipelines_ mean?

In the *Data Processing Pipelines* charter, it seemed like the GG rather than the AWG was responsible for pipeline approval? Does _review_ imply _review and approve_?

I'm having difficult parsing the second sentence. I don't understand the reference to _Changes to these ... pipelines_ and _comment on what are better ..._.

File: charters/UnityBenchmarkingService/charter.md [hunk 24]
```diff
@@ -0,0 +1,59 @@
...
+## In-scope
+
+* Benchmarking for standardized HCA data processing pipelines will be automated and included in Unity
+* Data used to run pipelines during benchmarking will be standardized and enforced to promote comparable analysis runs
+* Benchmarking data can change without notice but will be consistent in file format and file schema with previous data. If file format or schema do change (due to updates in the DCP or in standardized data processing pipelines) notice will be given to users
+* Leaderboards showing how pipelines that process the same assay differ in performance will be created as different pipelines for the same assay are collected
+* Documentation for the requirements of pipelines to be benchmarked by the Unity Benchmarking Service
+* Documentation on benchmarking process and outputs
+
+### Scientific "guardrails"
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md:25]: I would use stronger normative language in this section such as must, should, etc.

File: charters/UnityBenchmarkingService/charter.md (hunk 16 above)
Comment (brianraymor) [reply charters/UnityBenchmarkingService/charter.md]: Perhaps he meant  _Metrics and their presentation [or use] in leaderboards must be reviewed_?

And this could be merged into the earlier *Leaderboards* bullet?

File: charters/UnityBenchmarkingService/charter.md [hunk 25]
```diff
@@ -0,0 +1,59 @@
...
+* Educational training or user assistance on how to create pipelines
+* Recommendations on how to improve pipelines based on benchmarking
+* Pipelines benchmarking community data processing pipelines are not required to be included in the Unity benchmarking service (but are allowed). Community members may benchmark community pipelines using data and file format standards in the system. If community pipelines process assays not currently processed by standard HCA pipelines or assays not being scoped by the team, setting up the system will require a significant amount of work and will be evaluated given team bandwidth
+
+## Milestones and Deliverables
+
+__2018:__ Infrastructure operational and secure, HCA data processing pipelines team has initiated using system with a standardized HCA data processing pipeline.   
+__2019:__ System updated to include all data processing pipelines operational as standardized HCA pipelines.   
+__2019:__ Leaderboards are created for standardized HCA pipelines.   
+
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md:41]: It would be helpful to have more granularity (Q1, H2) for the **2019** milestone dates.

File: charters/UnityBenchmarkingService/charter.md [hunk 26]
```diff
@@ -0,0 +1,59 @@
...
+## Out-of-scope
+
+* Educational training or user assistance on how to create pipelines
+* Recommendations on how to improve pipelines based on benchmarking
+* Pipelines benchmarking community data processing pipelines are not required to be included in the Unity benchmarking service (but are allowed). Community members may benchmark community pipelines using data and file format standards in the system. If community pipelines process assays not currently processed by standard HCA pipelines or assays not being scoped by the team, setting up the system will require a significant amount of work and will be evaluated given team bandwidth
+
+## Milestones and Deliverables
+
+__2018:__ Infrastructure operational and secure, HCA data processing pipelines team has initiated using system with a standardized HCA data processing pipeline.
```
Comment (brianraymor) [inline charters/UnityBenchmarkingService/charter.md]: HCA data processing pipelines team -> HCA DCP Data Processing Pipelines team

BIt of trouble parsing _has initiating using system_

File: charters/UnityBenchmarkingService/charter.md (hunk 17 above)
Comment (brianraymor) [reply charters/UnityBenchmarkingService/charter.md]: I agree with @lauraclarke that the language could be clarified or you could remove the warnings and simply state the community pipelines are completely out of scope.

It's sometimes hard to parse tongue twisters like _community data processing pipelines_ or _community pipelines process assays_.  Perhaps bolding or italicizing defined terms might help. Also be consistent about just using **Community Pipelines** as defined in similar charters.

**Review by brianraymor — CHANGES_REQUESTED** [review]

File: charters/UnityBenchmarkingService/charter.md (hunk 16 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: I like your wording  Brian, added.

Having the bullet here calls it out as a guardrail and centralizes them in one place.

File: charters/UnityBenchmarkingService/charter.md (hunk 17 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: I will try again, I think this is an important point.

File: charters/UnityBenchmarkingService/charter.md (hunk 18 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: Capitalized, I can pull in some definitions.

File: charters/UnityBenchmarkingService/charter.md (hunk 19 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md:16]: It enforces the standards, pipelines will not run in the system without complying with a minimal set of standards that are consistent with the DCP.

Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md:16]: Updated the language updates.

File: charters/UnityBenchmarkingService/charter.md (hunk 20 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md:18]: Updated to do this for as many as I thought made sense. Feel free to ping specific ones that are left if you feel like they should be changed.

File: charters/UnityBenchmarkingService/charter.md (hunk 21 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: ok  removed

File: charters/UnityBenchmarkingService/charter.md (hunk 22 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: nice, thanks!

File: charters/UnityBenchmarkingService/charter.md (hunk 24 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md:25]: ok, updated

File: charters/UnityBenchmarkingService/charter.md (hunk 26 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: will try again.

File: charters/UnityBenchmarkingService/charter.md (hunk 17 above)
Comment (brianraymor) [reply charters/UnityBenchmarkingService/charter.md]: Attempting a clarification:

_Pipelines that benchmark **Community Pipelines** are not required to be included in the Unity benchmarking service (but are allowed). If a **Community Pipeline** is related to an assay that is neither currently processed by a standardized HCA pipeline nor is being scoped by the Data Processing Pipelines team, then setting up the benchmark will require significant investments which are evaluated based on team priorities. Otherwise, community members may also benchmark *Community Pipelines* using data and file format standards in the system._

Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: Brian, nicely done here. Thank you for taking the time to help me through the wording,  wish I could have said it so well. Will use as is.

File: charters/UnityBenchmarkingService/charter.md [hunk 27]
```diff
@@ -0,0 +1,64 @@
+# Unity Benchmarking Service
+
+## Description
+
+The Unity Benchmarking Service automates the process of running and benchmarking data processing pipelines.
+
+## Objectives
+
+The Unity Benchmarking Service automate benchmarking for data processing pipelines, enforce standards for consistent reporting for historic comparison and creates opportunity to demonstrate improvements to data processing pipelines using a common benchmarking infrastructure. The Unity Benchmarking Service is used by developers of data processing pipelines for the HCA data processing service and is hosted in a public-facing manner; so anyone can benchmark their data processing pipelines against benchmarked pipelines in the DCP.
```
Comment (mweiden) [inline charters/UnityBenchmarkingService/charter.md]: 1. automates ... enforces
1. suggest `publicly hosted`
1. define acronyms
1. Linking to the [Data Processing Pipelines Charter](https://github.com/HumanCellAtlas/dcp-community/pull/35) in the last sentence might make the relationship between Unity and Data Processing Pipelines clearer

File: charters/UnityBenchmarkingService/charter.md [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+
+## Definitions
+
+__Community pipeline:__ A pipeline that is not officially used as a standard in the HCA project, but runs in the DCP and is operated by the Data Processing Service to enable community members to work with a diversity of submitted data until they have a standardized pipeline. These pipelines are not as vetted as standardized pipelines.  
+__HCA Analysis Working Group (AWG):__ A group of domain specialists that advise on and monitor scientific choices made in the Human Cell Atlas Project.  
+__Standardized Pipeline:__ An official pipeline of the HCA project.
```
Comment (mweiden) [inline charters/UnityBenchmarkingService/charter.md:15]: See comments on the above two points in the Data Processing Pipelines charter:

* [Re. AWG](https://github.com/HumanCellAtlas/dcp-community/pull/35#discussion_r236782001)
* [Re. Standardized Pipeline](https://github.com/HumanCellAtlas/dcp-community/pull/35#discussion_r236798180)

File: charters/UnityBenchmarkingService/charter.md [hunk 29]
```diff
@@ -0,0 +1,64 @@
...
+* Metrics and their use in leaderboards must be reviewed by the HCA scientific leadership (Analysis Working Group).
+
+## Out-of-scope
+
+* Educational training or user assistance on how to create pipelines
+* Recommendations on how to improve pipelines based on benchmarking
+* Pipelines that benchmark Community Pipelines are not required to be included in the Unity benchmarking service (but are allowed). If a Community Pipeline is related to an assay that is neither currently processed by a standardized HCA pipeline nor is being scoped by the Data Processing Pipelines team, then setting up the benchmark will require significant investments which are evaluated based on team priorities. Otherwise, community members may also benchmark Community Pipelines using data and file format standards in the system.
```
Comment (mweiden) [inline charters/UnityBenchmarkingService/charter.md:34]: For the last bullet, I'd eliminate the second sentence and revise for clarity. Example below:

`Unity does not require pipelines that benchmark community analysis pipelines, but allows community members to submit their own if they conform to Unity's data and file format standards.`

**Review by mweiden — COMMENTED** [review]
Just some comments for clarity and accessibility here

File: charters/UnityBenchmarkingService/charter.md (hunk 27 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md]: Updated all the above, note the current link will not work because I needed to use the link for the charter after it is reviewed and pulled in.

File: charters/UnityBenchmarkingService/charter.md (hunk 28 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md:15]: Addressed in that charter.

File: charters/UnityBenchmarkingService/charter.md (hunk 29 above)
Comment (TimothyTickle) [reply charters/UnityBenchmarkingService/charter.md:34]: It is more than this, just because the pipelines confirm to the data and file format standards does not mean that pipeline will be up in Unity, it is per assay and requires having standard data and benchmarking scripts. It is more than running the pipeline. Happy to hear an additional suggestion.

**Review by brianraymor — APPROVED** [review]

**Comment by mweiden** [comment]
Approved in the 2019-01-18 Architecture Call
